src/machine/parsed_results.rs

```diff
@@ -284,73 +287,79 @@ impl TryFrom<String> for QueryResolutionLine {
     }
 }
 
-fn split_nested_list(input: &str) -> Vec<String> {
+fn split_nested_list(input: &str) -> Vec<&str> {
     let mut level = 0;
     let mut start = 0;
     let mut result = Vec::new();
 
-    for (i, c) in input.chars().enumerate() {
+    let mut end = 0;
+    for c in input.chars() {
         match c {
             '[' => level += 1,
             ']' => level -= 1,
             ',' if level == 0 => {
-                result.push(input[start..i].trim().to_string());
-                start = i + 1;
+                result.push(input[start..end].trim());
+                start = end + ','.len_utf8();
             }
             _ => {}
         }
+        end += c.len_utf8();
     }
 
-    result.push(input[start..].trim().to_string());
+    result.push(input[start..].trim());
     result
 }
 
-impl TryFrom<String> for Value {
-    type Error = ();
-    fn try_from(string: String) -> Result<Self, Self::Error> {
+pub(crate) fn strip_delimiters(haystack: &str, prefix: char, suffix: char) -> Option<&str> {
+    haystack.strip_prefix(prefix)?.strip_suffix(suffix)
+}
+
+pub(crate) fn strip_delimiters_str<'hay>(haystack: &'hay str, prefix: &str, suffix: &str) -> Option<&'hay str> {
+    haystack.strip_prefix(prefix)?.strip_suffix(suffix)
+}
```

Comment:
I had to duplicate this as the Pattern trait is still unstable so we can't use it in generic bounds and as such can't abstract over the pattern type i.e. `char` and `&str` like the standard library can